### ISSUE TITLE
require clamping of index for OpVectorExtractDyanmic, OpVectorInsertDynamic, and bitfield insert and extract

### ISSUE THREAD

**Issue by dneto0** [gpuweb/spirv-execution-env#27]
This eliminates a source of undefined behaviour.

**Comment by Kangz** [comment]
Maybe I misunderstand the title, shouldn't this be handled by implementations as a code transform instead? I'm not sure how easy it is to spec that the application has to do it.

**Comment by dneto0** [comment]
Yes, the implementation has to ensure that these instructions behave as if clamped.  The shader coming into the implementation doesn't have to do anything special to get this behaviour.
